### ISSUE TITLE
Nest login (non-Oauth device type) fails after a change happened at Google's end

### ISSUE THREAD

**Issue by bobMuk** [domoticz/domoticz#3729]
Using the "Nest Thermostat/Protect" (ie. not Oauth) device type, this week logins have started to fail, looks like Google have changed something at the API side.

I've identified the issue, in Nest.cpp the user-agent header value is no longer being accepted by the API.  Changing it from:
`"user-agent:Nest/1.1.0.10 CFNetwork/548.0.4"`
...to:
`"User-Agent: Nest/3.0.1.15 (iOS) os=6.0"`
...fixes the issue.

Unfortunately I don't currently have the means to extract/recompile/upload the change (I tested it by hexedit'ing the binary).

**Comment by gizmocuz** [comment]
Thanks for this ! You could check 'fork' the code, checkout your forked copy, make changes, commit and push them, them make a pull request (PR)
I have changed the agent in beta 11487, hopefully it works again...

**Comment by FilipDem** [comment]
Has somebody already feedback of this correction? I just installed the beta version. To be sure, I checked with a hex editor if really the change mentioned above is included (and it is). However I still got the error "Nest: Invalid data received, or invalid username/password!". 

Should this fix also solve the problem when having migrated (accidentically) the original NEST account to account to Google? I did this accidentically when installing my new mobile phone.
